src/hotspot/share/gc/shenandoah/shenandoahAsserts.hpp

```diff
@@ -213,6 +232,10 @@ class ShenandoahAsserts {
 #define shenandoah_assert_heaplocked()
 #define shenandoah_assert_not_heaplocked()
 #define shenandoah_assert_heaplocked_or_safepoint()
+#define shenandoah_assert_control_or_vm_thread()
+#define shenandoah_assert_control_or_vm_thread_at_safepoint()
+#define shenandoah_assert_generational()
+#define shenandoah_assert_generations_reconciled()                                                             \
```

Comment:
There seems to be dangling `\` on this line.

Reply:
Good catch.

Reply:
https://bugs.openjdk.org/browse/JDK-8344151